File: seminar_1/task_3/main.py
class ListNode:
	def __init__(self, x):
	       	self.val = x
          	self.next = None

def has_cycle(head: ListNode) -> bool:
        fast = slow = head
        while fast:
            slow = slow.next
            fast = fast.next
            if fast:
                fast = fast.next
            else:
                return False

            if fast == slow:
                return True
        return False

File: seminar_1/task_3/main.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func cicleLinkedList(head *ListNode) bool {
	var (
		slow = head
		fast = head
	)
	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next.Next
		if slow == fast {
			return true
		}
	}
	return false
}

func main() {
	fmt.Println("task_3")
}
